src/components/App/App.js

```diff
@@ -9,28 +9,75 @@ import { api } from "../../utils/newsApi.js";
 import RegisterModal from "../Modals/RegisterModal/RegisterModal.js";
 import SuccessModal from "../Modals/SuccessModal/SuccessModal.js";
 import LoginModal from "../Modals/LoginModal/LoginModal.js";
+import { userApi } from "../../utils/mainApi.js";
+import { getToken, removeToken, setToken } from "../../utils/token.js";
+import { CurrentUserContext } from "../../contexts/CurrentUserContext.js";
+import ProtectedRoute from "../ProtectedRoute/ProtectedRoute.js";
 
 function App(props) {
   //#region Methods
 
   function getNews(query) {
     if (query) {
+      setKeyword(query);
       setIsSearching(true);
-      api.getNews(query)
-        .then((json) => {
-          setNews(json["articles"]);
+      api.getNews(query).then((newJson) => {
+          setNews(newJson["articles"]);
           setIsSearching(false);
         });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**